src/compiler/checker.ts

```diff
@@ -4333,6 +4338,10 @@ module ts {
             return !!getPropertyOfType(type, "0");
         }
 
+        function isTupleType(type: Type) : boolean {

```

Comment:
Ok, please put a comment explaining that in general, one should use isTupleLikeType for type system checks, but when someone depends on the existence of .elementTypes, call isTupleType. It has different semantics in from isTupleLikeType because it has to be literally written as a tuple, and not as an object with a '0' property.